### PR TITLE
introduce CHECK_INC/LIBS in tests/Makefile

### PR DESCRIPTION
**Pull request by drossetti**
mainly to deal with cases when libbcheck is installed in a non-system dir

### CODE REVIEW

**Comment by pakmarkthub** [comment]
libcheck comes with check.pc. Users can use `PKG_CONFIG_PATH=<folder-that-has-check.pc> make`. That would do the trick when they want to use libcheck installed in a custom location.

**Comment by drossetti** [comment]
Interesting so it is simply a matter of documenting the PKG_CONFIG_PATH=<folder-that-has-check.pc> trick.
thanks

**Comment by drossetti** [comment]
@pakmarkthub still wouldn't we need the INCLUDE part of my patch?

File: tests/Makefile [hunk 1]
```diff
@@ -1,13 +1,16 @@
...
+CHECK_LIBS ?= $(shell pkg-config --libs check)
+
 GDRAPI_INC := ../include
 GDRAPI_SRC := ../src
 
 CUDA_LIB := -L $(CUDA)/lib64 -L $(CUDA)/lib -L /usr/lib64/nvidia -L /usr/lib/nvidia
 CUDA_INC += -I $(CUDA)/include
 
-CPPFLAGS := $(CUDA_INC) -I $(GDRAPI_INC) -I $(GDRAPI_SRC) -I $(CUDA)/include
+CPPFLAGS := $(CUDA_INC) -I $(GDRAPI_INC) -I $(GDRAPI_SRC) -I $(CUDA)/include -I $(CHECK_INC)
```
Comment (pakmarkthub) [inline tests/Makefile]: pkg-config --cflags-only-I returns "-I". So, no need for the leading "-I".

File: tests/Makefile [hunk 2]
```diff
@@ -1,13 +1,16 @@
 DESTBIN ?= 
 CUDA ?= /usr/local/cuda
 
+CHECK_INC  ?= $(shell pkg-config --cflags-only-I check)
```
Comment (pakmarkthub) [inline tests/Makefile]: To help users who are not familiar with pkg-config, I would prefer to have CHECK_PATH possibly set by the users. If we detect CHECK_PATH, we use it to compose CHECK_INC and CHECK_LIBS. If not, get the flags from pkg-config.

Comment (drossetti) [reply tests/Makefile]: understood. I'll convert these assignments  into :=

**Comment by drossetti** [comment]
see #118 for a related problem

**Comment by drossetti** [comment]
@pakmarkthub I'd appreciate if you could take a look

**Comment by pakmarkthub** [comment]
@drossetti 

LGTM.

Two small comments:

1. In README, could you also mention about using PKG_CONFIG_PATH with the packaging scripts such as `$ CUDA=<cuda-install-top-dir> PKG_CONFIG_PATH=<...> ./build-deb-packages.sh`.

2. While you are at it, can you make `build-rpm-packages.sh` support `PKG_CONFIG_PATH` as well?

**Comment by drossetti** [comment]
> 1. In README, could you also mention about using PKG_CONFIG_PATH with the packaging scripts such as `$ CUDA=<cuda-install-top-dir> PKG_CONFIG_PATH=<...> ./build-deb-packages.sh`.

I'm not sure that would be enough, PKG_CONFIG_PATH would need to be passed down to rpmbuild somehow, most probably by extending the spec file.

> 2. While you are at it, can you make `build-rpm-packages.sh` support `PKG_CONFIG_PATH` as well?

Same for this case, I'm not convinced it is so easy.

And in the end, if the user is going to build the packages, because the dependency on libcheck is in their meta-data, PKG_CONFIG_PATH would also need to be complemented by forcing the build dependency first, and maybe the installation dependency later.

So I'd rather take more time to think this through.

**Pull request merged**